app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -187,6 +188,17 @@ def process_availability(availability)
         end
       end
 
+      def parse_availability(availabilities, resources_by_metric_id)
+        processed_availabilities_ids = availabilities.map do |availability|
```

Comment:
+1 this is much readable to me than using `.map!` Also `.map!` doesn't sound very "functional"